kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -787,14 +789,21 @@ internal class CodeWriter(
       val aliasNames = mutableMapOf<String, T>()
       var segmentsToUse = 0
       // Iterate until we have unique aliases for all names.
-      while (aliasNames.size != canonicalNamesToQualifiedNames.size) {
+      outer@ while (aliasNames.size != canonicalNamesToQualifiedNames.size) {
         segmentsToUse += 1
         aliasNames.clear()
         for ((segments, qualifiedName) in canonicalNameSegmentsToQualifiedNames) {
           val aliasPrefix = segments.takeLast(min(segmentsToUse, segments.size))
             .joinToString(separator = "")
             .replaceFirstChar { if (!capitalizeAliases) it.lowercaseChar() else it }
-          val aliasName = aliasPrefix + simpleName.replaceFirstChar(Char::uppercaseChar)
+          val aliasSuffix = "_".repeat((segmentsToUse - segments.size).coerceAtLeast(0))
+          val aliasName = aliasPrefix + simpleName.replaceFirstChar(Char::uppercaseChar) + aliasSuffix
+          // If this name has already been imported (e.g. a regular import already exists with this name),
+          // continue trying with a greater number of segments.
```

Comment:
It seems that we do this for all names if we encounter a conflict for just one of them, as demonstrated by the `importAliasCollisionWithImportWhenRegularImportComputedFirst` test: if we linked `com.squareup.meal.TacoMeal` to `TacoMeal`, we'll generate `import com.squareup.burrito.Meal as SquareupBurritoMeal` and `import com.squareup.taco.Meal as SquareupTacoMeal` imports, even though `import com.squareup.burrito.Meal as BurritoMeal` would've been a shorter non-conflicting alias import. Can we maybe track `canonicalNamesToQualifiedNames` we've found non-conflicting imports for, and only try a larger number of segments for those that are still conflicting?

Reply:
Yup that's true. I think for this one I'd like to leave it as-is if that's alright with you since optimizing this part is a little more involved than what you pointed out in the first comment. I think this scenario should be rare enough that it's probably not likely to make that much of a difference.